### PR TITLE
[Validator] Add the option `filenameMaxLength` to the File constraint

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fixes https://github.com/symfony/symfony-docs/issues/17953

### CODE REVIEW

**Review by mdoutreluingne — APPROVED** [review]

File: reference/constraints/File.rst [hunk 1]
```diff
@@ -268,6 +268,30 @@ You can find a list of existing mime types on the `IANA website`_.
...
+
+**type**: ``string`` **default**: ``The filename is too long. It should have {{ filename_max_length }} character or less.|The filename is too long. It should have {{ filename_max_length }} characters or less.``
+
+.. versionadded:: 6.3
+
+    The ``filenameTooLongMessage`` was introduced in Symfony 6.3.
+
+The message displayed if the filename of the file exceeds the limit set
+with the ``filenameMaxLength`` option.
+
```
Comment (ke20) [inline reference/constraints/File.rst:294]: I think you can add :
```rst
You can use the following parameters in this message:

==============================  ==============================================================
Parameter                       Description
==============================  ==============================================================
``{{ filename_max_length }}``   Maximum number of characters allowed
==============================  ==============================================================
```

I saw some similar thing for other options.

Thanks for the PR :)

**Review by ke20 — APPROVED** [review]

File: reference/constraints/File.rst (hunk 1 above)
Comment (alexandre-daubois) [reply reference/constraints/File.rst:294]: Oooh yeah definitely. Thank you!

File: reference/constraints/File.rst [hunk 2]
```diff
@@ -268,6 +268,38 @@ You can find a list of existing mime types on the `IANA website`_.
     (i.e. the form type is not defined explicitly in the ``->add()`` method of
     the form builder) and when the field doesn't define its own ``accept`` value.
 
+``filenameMaxLength``
+~~~~~~~~~~~~~~~~~~~~~
+
+**type**: ``int`` **default**: ``null``
```
Comment (ke20) [inline reference/constraints/File.rst]:
```suggestion
**type**: ``integer`` **default**: ``null``
```

Comment (alexandre-daubois) [reply reference/constraints/File.rst]: Thanks!

Comment (OskarStark) [reply reference/constraints/File.rst]: This should be spotted by DOCtor-RST 🤔 , was this the case?

Comment (alexandre-daubois) [reply reference/constraints/File.rst]: Just checked. It was spotted, I just missed it :smile: 

https://github.com/symfony/symfony-docs/actions/runs/4245728234/jobs/7381615847

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Alexandre.

**Pull request merged**